src/vmax.rs

```diff
@@ -1188,3 +1187,190 @@ impl ChildPoint for PhysicalCapacity {
         p.add_field("total_capacity_gb", TsValue::Float(self.total_capacity_gb));
     }
 }
+
+#[derive(Debug, Deserialize, IntoPoint)]
+pub struct Volume {
+    #[serde(rename = "volumeId")]
+    pub volume_id: String,
+    #[serde(rename = "type")]
+    pub volume_type: Option<String>,
+    pub emulation: Option<String>,
+    pub ssid: Option<String>,
+    pub allocated_percent: Option<u64>,
+    pub cap_gb: Option<f64>,
+    pub cap_mb: Option<f64>,
+    pub cap_cyl: Option<u64>,
+    pub status: Option<String>,
+    pub reserved: Option<bool>,
+    pub pinned: Option<bool>,
+    pub physical_name: Option<String>,
+    pub volume_identifier: Option<String>,
+    pub wwn: Option<String>,
+    pub encapsulated: Option<bool>,
+    pub num_of_storage_groups: Option<u64>,
+    pub num_of_fron_end_paths: Option<u64>,
+    #[serde(rename = "storageGroupId")]
+    pub storage_group_id: Option<Vec<String>>,
+    // ignoring these two fields as they are not needed,
+    // and don't have to store an array unnecessarily
+    //#[serde(rename = "symmetrixPortKey")]
+    //pub symmetrix_port_key: Option<Vec<SymmetrixPortKey>>,
+    //#[serde(rename = "rdfGroupId")]
+    //pub rdf_group_id: Option<Vec<u64>>,
+    pub snapvx_source: Option<bool>,
+    pub snapvx_target: Option<bool>,
+    pub cu_image_base_address: Option<String>,
+    pub has_effective_wwn: Option<bool>,
+    pub effective_wwn: Option<String>,
+    pub encapsulated_wwn: Option<String>,
+}
+
+#[test]
+fn test_get_vmax_json_volume() {
+    use std::fs::File;
+    use std::io::Read;
+
+    let mut f = File::open("tests/vmax/slo_volume.json").unwrap();
+    let mut buff = String::new();
+    f.read_to_string(&mut buff).unwrap();
+
+    let i: Volume = serde_json::from_str(&buff).unwrap();
+    println!("result: {:#?}", i);
+    println!("point: {:#?}", i.into_point(Some("vmax_slo_volume")));
```

Comment:
This is going to break when you rebase against master.  It has another param to say whether it's a time series dataset or not.